File: src/main/java/um/haberes/core/model/dto/FormularioAsignacionCargo.java
package um.haberes.core.model.dto;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import um.haberes.core.hexagonal.facultad.domain.model.Facultad;
import um.haberes.core.hexagonal.liquidaciones.categoria.domain.model.Categoria;
import um.haberes.core.hexagonal.personas.dependencia.domain.model.Dependencia;

import java.util.List;

@Getter
@Setter
@EqualsAndHashCode
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class FormularioAsignacionCargo {

    private List<Categoria> categorias = null;

    private List<Categoria> categoriasAsignables = null;

    private List<Dependencia> dependencias = null;

    private List<Facultad> facultades = null;
}

File: src/main/java/um/haberes/core/kotlin/model/dto/FormularioAsignacionCargo.kt
package um.haberes.core.kotlin.model.dto

import um.haberes.core.hexagonal.facultad.domain.model.Facultad
import um.haberes.core.kotlin.model.Categoria
import um.haberes.core.kotlin.model.Dependencia
import um.haberes.core.hexagonal.facultad.infrastructure.persistence.entity.FacultadEntity

data class FormularioAsignacionCargo(

    var categorias: List<Categoria?>? = null,
    var categoriasAsignables: List<Categoria>? = null,
    var dependencias: List<Dependencia>? = null,
    var facultades: List<Facultad>? = null

)
